logpuzzle.py

```diff
@@ -29,7 +32,12 @@ def read_urls(filename):
     Screens out duplicate urls and returns the urls sorted into
     increasing order."""
     # +++your code here+++
-    pass
+    head, _, tail = filename.partition("_")
```

Comment:
Unnecessarily assigning values to head and '_' when using .partition(). Consider using index slicing instead:

Instead of:
    head, _, tail = filename.partition('_')

Consider:
    host = filename[filename.index('_') + 1:]
Or:
    _, _, tail = filename.partition()